pkg/provider/vault/symlink.go

```diff
@@ -0,0 +1,43 @@
+package vault
+
+import (
+	"context"
+	"regexp"
+	"strings"
+)
+
+func isSymlink(secret any) bool {
```

Comment:
Since I am not yet very familiar with the type `any`, do we need any type constraint in case the function receives a non-string type? Is there any downside in using directly a `string` type here?

Reply:
The secret retrieve from vault is of type `map[string]interface{}` type any is the same as `interface{}`. Added check for type conversion to `string`